Junk Codes/Ultrasonic.ino

```diff
@@ -1,138 +1,124 @@
-  delay(5000);
+void getValues()
+{
+digitalWrite(trigPin, LOW);
+delayMicroseconds(2);
+digitalWrite(trigPin, HIGH);
+delayMicroseconds(10);
+digitalWrite(trigPin, LOW);
+duration = pulseIn(echoPin, HIGH);
+dist = duration * 0.034 / 2;
+Serial.print("Distance:");
+Serial.println(dist);
+if(dist>=10)
+{
+steppermotor.setSpeed(700); 
+steppermotor.step(StepsRequired);
+deg=StepsRequired/360;
+Serial.println("angle");
+Serial.println(deg);
+}
 }
+
+
 void reconnect()
 {
-  // Loop until we're reconnected
-  //------------How to restart Nodemcu
-  int counter = 0;
-  while (!client.connected())
-  {
-    //    if (counter == 5)
-    //    {
-    //      ESP.restart();
-    //    }
-    //    counter += 1; wait i will do
-    Serial.print("Attempting MQTT connection...");
-    // Attempt to connect
-
-    if (client.connect(mqttClientName, mqttUser, mqttPassword))
-    {
-      Serial.println("connected");
-    }
-    else
-    {
-      Serial.print("failed, rc=");
-      Serial.print(client.state());
-      Serial.println(" try again in 5 seconds");
-      // Wait 5 seconds before retrying
-      delay(5000);
-    }
-  }
+int counter = 0;
+while (!client.connected())
+{
+if (counter == 5)
+{
+ESP.restart();
 }
-
-void loop()
+counter += 1;
+Serial.print("Attempting MQTT connection...");
+if (client.connect(mqttClientName, mqttUser, mqttPassword))
 {
-  if (!client.connected())
-  {
-    reconnect();
-  }
-  client.publish(mqttLatExt, String(latExt).c_str(), true); // CHange respective topics
-  client.publish(mqttLngExt, String(lngExt).c_str(), true); //---do--
-  delay(1000);
-
-  for (int i = 0; i <= 360; i++)
-  {
-    radarServo.write(i);
-    delay(50);
-    digitalWrite(trigPin, LOW);
-    delayMicroseconds(2);
-    digitalWrite(trigPin, HIGH);
-    delayMicroseconds(10);
-    digitalWrite(trigPin, LOW);
-    duration = pulseIn(echoPin, HIGH);
-    distinCM = duration * 0.034 / 2;
-    Serial.print(i);
-    Serial.print("");
-    Serial.print(distinCM);
-    Serial.print("#");
-  }
-  for (int i = 360; i >= 0; i--)
-  {
-    radarServo.write(i);
-    delay(50);
-    digitalWrite(trigPin, LOW);
-    delayMicroseconds(2);
-    digitalWrite(trigPin, HIGH);
-    delayMicroseconds(10);
-    digitalWrite(trigPin, LOW);
-    duration = pulseIn(echoPin, HIGH);
-    distinCM = duration * 0.034 / 2;
-    Serial.print(i);
-    Serial.print("");
-    Serial.print(distinCM);
-    Serial.print("#");
-  }
+Serial.println("connected");
+}
+else
+{
+Serial.print("failed, rc=");
+Serial.print(client.state());
+Serial.println(" try again in 5 seconds");
+delay(5000);
+}
+}
 }
 
 
-// Format to send data to mqtt
-//{"deg": 180,"dis":30}
-//sprintf(data, "{"deg": %s,"dis":%s}", degree, distance);
-//Use stepper motor
+void loop()
+{
+if (!client.connected())
+{
+reconnect();
+}
+getValues();
+client.publish(mqttRadar,String(radarData).c_str(),true);
+delay(1000);
+sprintf(radarData,"{"deg":"%s","dist":"%s"}",ultrangle,ultradist);
```

Comment:
`client.publish(mqttRadar,String(radarData).c_str(),true);`
The above code publishes `radarData` to mqtt topic but below code write the data to `radarData`. The thing is you are sending the data to mqtt before writing it. `sprintf` should come first
`sprintf(radarData,"{"deg":"%s","dist":"%s"}",ultrangle,ultradist);`